src/rgb_primaries.rs

```diff
@@ -0,0 +1,101 @@
+//!This module defines the red, blue and green primaries for the comman Rgb color spaces
+use num::Float;
+
+
+use {Yxy, Xyz, Rgb, IntoColor};
+use white_point::{WhitePoint, D65};
+use chromatic_adaptation::AdaptInto;
+use matrix::{Mat3, matrix_inverse, multiply_xyz_to_rgb};
+use flt;
+
+///Represents the tristimulus values for the Rgb primaries
+pub trait RgbPrimaries<Wp, T>
+    where T:Float,
+        Wp: WhitePoint<T>,
+{
+    ///Tristimulus values for red
+    fn red() -> Yxy<Wp, T>;
+    ///Tristimulus values for green
+    fn green() -> Yxy<Wp, T>;
+    ///Tristimulus values for blue
+    fn blue() -> Yxy<Wp, T>;
+}
+
+///Srgb color space with default D65 white point
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct SrgbSpace;
+
+impl<T: Float> RgbPrimaries<D65, T> for SrgbSpace {
```

Comment:
This is where `D65` can be replaced by a generic `Wp`. The way I understand it is that the primaries are the same, no matter what the white point is.